site_scons/buildutils.py

```diff
@@ -34,6 +36,396 @@
     TPathLike = TypeVar("TPathLike", Path, str)
 
 
+class Option:
+    """Object corresponding to SCons configuration option"""
+
+    def __init__(self,
+            name: str,
+            description: str,
+            default: "Union[bool, str]",
+            choices: list = None):
+        self.name = name
+        self.description = Option._deblank(description)
+        self.default = default
+        self.choices = choices
+
+        self._wrapper = textwrap.TextWrapper(width=80)
+        self.set_wrapper_indent(4)
+
+    @property
+    def parameters(self) -> "Union[Tuple[str, str, str, str], Tuple[str, str, str]]":
+        """Parameters of the SCons configuration option"""
+        if self.choices is None:
+            return self.name, self.description, self.default
+        return self.name, self.description, self.default, self.choices
+
+    def to_scons(self, env: "SCEnvironment" = None) -> "SCVariables":
```

Comment:
`"Optional[SCnvironment]" ` is equivalent to `"Union[SCEnvironment, None]"`, so it is required here, and everywhere `None` can be passed. I agree its not a great syntax, and starting with 3.10, we can use `|` instead of `Union`, so this would become `"SCEnvironment | None" `.

Reply:
Hm. So anytime we pass `None` as a sentinel value, it is required to add this. I routinely use `None` as default, which I should probably stop then.

Reply:
I can live with `"Union[SCEnvironment, None]"` ... extrapolating the `Optional[...]` would apply to every single keyword argument!

Reply:
I think the preference is generally for `Optional[value]` when `None` is used as a sentinel value (indicating that there's some default behavior), and the `Union` is for when `None` is a valid value. Obviously that's not enforced at a type-checking level, but IMO it communicates intent more clearly.

`Optional` is only required for cases where `None` is the default value (or a possible value). In cases where there is another type as the default, that needs to be in the `Union` instead, and you have to use `Union` 😊

Reply:
I guess I learned something new here - thanks for explaining! Will change to `Optional[...]` then. I replaced all other occurrences of `None` as I really don't like the clutter it creates.